aws-iot-dimension/src/test/java/com/amazonaws/iot/dimension/UpdateHandlerTest.java

```diff
@@ -0,0 +1,215 @@
+package com.amazonaws.iot.dimension;
+
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_ARN;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_NAME;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_TYPE;
+import static com.amazonaws.iot.dimension.TestConstants.DIMENSION_VALUE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.services.iot.model.InternalFailureException;
+import software.amazon.awssdk.services.iot.model.InvalidRequestException;
+import software.amazon.awssdk.services.iot.model.IotRequest;
+import software.amazon.awssdk.services.iot.model.TagResourceRequest;
+import software.amazon.awssdk.services.iot.model.UntagResourceRequest;
+import software.amazon.awssdk.services.iot.model.UpdateDimensionRequest;
+import software.amazon.awssdk.services.iot.model.UpdateDimensionResponse;
+import software.amazon.cloudformation.exceptions.CfnInternalFailureException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+@ExtendWith(MockitoExtension.class)
+public class UpdateHandlerTest {
+
+    protected static final software.amazon.awssdk.services.iot.model.Tag PREVIOUS_SDK_RESOURCE_TAG =
+            software.amazon.awssdk.services.iot.model.Tag.builder()
+                    .key("PreviousTagKey")
+                    .value("PreviousTagValue")
+                    .build();
+    protected static final software.amazon.awssdk.services.iot.model.Tag DESIRED_SDK_RESOURCE_TAG =
+            software.amazon.awssdk.services.iot.model.Tag.builder()
+                    .key("DesiredTagKey")
+                    .value("DesiredTagValue")
+                    .build();
+
+    @Mock
+    private AmazonWebServicesClientProxy proxy;
+
+    @Mock
+    private Logger logger;
+
+    @Spy
+    private UpdateHandler handler;
+
+    @Test
```

Comment:
Case handling where the resource we are trying to update is already deleted out of band

Reply:
Added a new unit test: `handleRequest_ResourceAlreadyDeleted_VerifyException`